lib/wallet/migrations.js

```diff
@@ -0,0 +1,301 @@
+ * https://github.com/handshake-org/hsd
+ */
+
+'use strict';
+
+const assert = require('bsert');
+const Logger = require('blgr');
+const Network = require('../protocol/network');
+const {
+  AbstractMigration,
+  MigrationResult,
+  Migrations,
+  types,
+  oldLayout
+} = require('../migrations/migrations');
+const MigrationState = require('../migrations/state');
+const layouts = require('./layout');
+const layout = layouts.wdb;
+
+/**
+ * Switch to new migrations layout.
+ */
+
+class MigrateMigrations extends AbstractMigration {
+  /**
+   * Create migrations migration.
+   * @param {ChainMigrations} options
+   */
+
+  constructor(options) {
+    super(options);
+
+    this.options = options;
+    this.logger = options.logger.context('wallet-migrations-migrate');
+    this.db = options.db;
+    this.ldb = options.ldb;
+    this.layout = options.layout;
+  }
+
+  async check() {
+    return types.MIGRATE;
+  }
+
+  /**
+   * Actual migration
+   * @param {Batch} b
+   * @returns {Promise}
+   */
+
+  async migrate(b) {
+    this.logger.info('Migrating migrations..');
+    const state = new MigrationState();
+    state.nextMigration = 1;
+
+    if (await this.ldb.get(oldLayout.M.encode(0))) {
+      b.del(oldLayout.M.encode(0));
+      state.nextMigration = 2;
+    }
+
+    this.db.writeVersion(b, 1);
+    b.put(this.layout.M.encode(), state.encode());
+  }
+
+  static info() {
+    return {
+      name: 'Migrate wallet migrations',
+      description: 'Wallet migration layout has changed.'
+    };
+  }
+}
+
+/**
+ * Run change address migration.
+ * Applies to WalletDB v0
+ */
+
+class MigrateChangeAddress extends AbstractMigration {
+  /**
+   * Create change address migration object.
+   * @constructor
+   * @param {WalletMigrations} options
+   */
+
+  constructor(options) {
+    super(options);
+
+    this.options = options;
+    this.logger = options.logger.context('change-address-migration');
+    this.db = options.db;
+    this.ldb = options.ldb;
+  }
+
+  /**
+   * Migration and check for the change address
+   * are done in the same step.
+   */
+
+  async check() {
+    return types.MIGRATE;
+  }
+
+  /**
+   * Actual migration
+   * @param {Batch} b
+   * @param {WalletMigrationResult} pending
+   * @returns {Promise}
+   */
+
+  async migrate(b, pending) {
+    const wids = await this.ldb.keys({
+      gte: layout.W.min(),
+      lte: layout.W.max(),
+      parse: key => layout.W.decode(key)[0]
+    });
+
+    let total = 0;
+    for (const wid of wids) {
+      const wallet = await this.db.get(wid);
+
+      this.logger.info('Checking wallet (id=%s, wid=%d).',
+                       wallet.id, wid);
+
+      total += await this.migrateWallet(b, wallet);
+    }
+
+    if (total > 0)
+      pending.rescan = true;
+  }
+
+  async migrateWallet(b, wallet) {
+    let total = 0;
+    for (let i = 0; i < wallet.accountDepth; i++) {
+      const account = await wallet.getAccount(0);
```

Comment:
```suggestion
      const account = await wallet.getAccount(i);
```

right?? That's how the original code for this migration looked

Reply:
totally !!! Great catch